packages/opencensus-exporter-trace-stackdriver/test/test-stackdriver.ts

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {types} from '@opencensus/opencensus-core';
+import {classes} from '@opencensus/opencensus-core';
+import {logger} from '@opencensus/opencensus-core';
+import * as assert from 'assert';
+import * as fs from 'fs';
+import * as mocha from 'mocha';
+import * as nock from 'nock';
+
+import {StackdriverExporterOptions, StackdriverTraceExporter} from '../src/';
+
+import * as nocks from './nocks';
+
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+let PROJECT_ID = 'fake-project-id';
+
+function checkEnvironoment(): boolean {
+  return false;
+}
+
+describe('Stackdriver Exporter', function() {
```

Comment:
My general belief here is that in this test file a `dryrun` flag is not needed, and we can _always_ use nocks to simulate endpoints. We can add a separate test (in, say, a `system-test` or `test/integration` directory) that then uses real network endpoints to ensure that the module works with the Stackdriver API. That way it's easy to identify and run all non-network tests at once, and separately run all network/authenticated tests at once.

Reply:
Hmm... I've thought about this a little more and I think it should be OK. That being said, I think `dryrun` should be coming from its own environmental variable -- maybe something like `OPENCENSUS_NETWORK_TESTS` -- so that we can easily "turn on" network accesses for tests in all packages with a single switch.

Reply:
That's a nice idea, just implemented that. Used `OPENCENSUS_NETWORK_TESTS` as you suggested.